### PR TITLE
Ignore insecure files and directories in compinit

### PR DESCRIPTION
**Pull request by maximbaz**
This might be a bit controversial PR, and I'm definitely open for suggestions on how to do things differently.

Let me explain the problem that I'm trying to solve. I have all my dotfiles (including antigen) in my home dir, all of them are owned by my regular user. Sometimes I know that I need to execute a few privileged operations, so instead of prepending each command with `sudo` I change to root with `$ sudo -s` and then execute the commands without any prefix.

I like using `$ sudo -s`, because it preserves my user environment, $HOME directory, my prompt, antigen configuration, everything stays the same. However, when I enter `$ sudo -s`, I'm welcomed with the following message:

```
❯ sudo -s
zsh compinit: insecure directories and files, run compaudit for list.
Ignore insecure directories and files and continue [y] or abort compinit [n]?
```

This happens because of the [following](http://zsh.sourceforge.net/Doc/Release/Completion-System.html):

> For security reasons compinit also checks if the completion system would use files not owned by root or by the current user, or files in directories that are world- or group-writable or that are not owned by root or by the current user.

Basically, compinit complains because current user is `root`, and dotfiles are owned by a different user.

I can't "fix" the permissions, I want my dotfiles to be owned by my own user, so I see this annoying warning every time.

Adding `-i` is an easy win, this flag makes compinit silently ignore all insecure files and directories.

What do you think about it, can we add it?

### CODE REVIEW

**Comment by desyncr** [comment]
I see no major issues to add it. May be we can create a config for it `ANTIGEN_COMPINIT` (as we already have `ANTIGEN_COMPDUMP`).

**Comment by maximbaz** [comment]
Great! Should the default value for `ANTIGEN_COMPINIT` be `compinit -i` or plain `compinit`?

**Comment by desyncr** [comment]
Should be `compinit` for `v2.2.3` if there is need for it we may change to `compinit -i` on `v2.4` or greater.

**Comment by desyncr** [comment]
It seems it doesn't like running `compinit -i` as `ANTIGEN_COMPINIT` value. I'll rather use `ANTIGEN_COMPINIT_OPTS`, see the other PR.

**Comment by desyncr** [comment]
Merged in #634. Thanks for looking into this!

**Comment by maximbaz** [comment]
#634 works perfectly, thank you very much!
